src/Core_Fix_ResourceUnloadOptimizations/Core.ResourceUnloadOptimizations.cs

```diff
@@ -94,6 +104,32 @@ private static void RunGarbageCollect()
             GC.Collect(GC.MaxGeneration);
         }
 
+        private static void RunForceCleanupIfNecessary()
+        {
+            _forceCleanup = 10;
+
+            var status = MemoryInfo.GetCurrentStatus();
+            if (status == null)
+                return;
+
+            if (status.dwMemoryLoad < PercentForceCleanupThreshold.Value)
+            {
+                _countForceCleanup = 0;
+                return;
+            }
+
+            if (++_countForceCleanup > 8)
+            {
+                // Cleanup was performed, but memory utilization remained high.
+                // Maybe the system memory is low or you have a huge scene open.
+                // Cleanup will only cause stuttering and should be aborted.
+                return;
+            }
+
+            GC.Collect(GC.MaxGeneration);
+            Resources.UnloadUnusedAssets();

```

Comment:
There is no need to call GC.Collect, Resources.UnloadUnusedAssets already does this internally.

Reply:
Erased.  

Heh, I didn't know that. I thought they were supposed to be written in pairs.

Reply:
You can read more about it here https://forum.unity.com/threads/resources-unloadunusedassets-vs-gc-collect.358597/#post-2321896